osquery/database/plugins/rocksdb.cpp

```diff
@@ -114,6 +120,41 @@ class RocksDBDatabasePlugin : public DatabasePlugin {
 /// Backing-storage provider for osquery internal/core.
 REGISTER_INTERNAL(RocksDBDatabasePlugin, "database", "rocksdb");
 
+class RocksDBGuard {
+ public:
+  RocksDBGuard() : my_tid_(platformGetTid()), locked_(false) {}
+
+  bool safeEnter() {
+    if (!owner_tid_is_valid_ || owner_tid_ != my_tid_) {
+      mu_.lock();
```

Comment:
Interesting, I think flipping the `forward_` back and forth may work. Or introduce another toggle? But I think forward makes sense, since it already means "should I be forwarding logs".